miden-lib/src/transaction/inputs.rs

```diff
@@ -290,3 +298,23 @@ fn add_input_notes_to_advice_inputs(
     // NOTE: keep map in sync with the `prologue::process_input_notes_data` kernel procedure
     inputs.extend_map([(tx_inputs.input_notes().commitment(), note_data)]);
 }
+
+// KERNEL HASHES INJECTOR
+// ------------------------------------------------------------------------------------------------
+
+/// Inserts kernel hashes and hashes of their procedures into the provided advice inputs.
+///
+/// Inserts the following entries into the advice map:
+/// - The accumulative hash of all kernels |-> array of each kernel hash.
+/// - The hash of the kernel `i` (for every `i`) |-> array of the current kernel's procedure hashes.
+pub fn add_kernel_hashes_to_advice_inputs(inputs: &mut AdviceInputs) {
+    let kernel_hashes = [TransactionKernel::kernel_hash().as_elements()].concat();
+
+    // insert kernels root with kernel hashes into the advice map
+    inputs.extend_map([(TransactionKernel::kernel_root(), kernel_hashes)]);
+
+    // insert kernel hash with procedure hashes into the advice map
+    // TODO: insert kernel procedures iterating over kernels
+    inputs
+        .extend_map([(TransactionKernel::kernel_hash(), TransactionKernel::procedures_as_felts())]);
```

Comment:
We actually care only about the procedures for the selected kernel - so, we won't need a loop here. But what we may need is to pass `kernel_offset` into this function. Then, it could be used to select an appropriate kernel. Though, maybe this is for another PR (worth updating the TODO though).